### ISSUE TITLE
Server list refresh is blocking th UI (all servers are queried in a single frame)

### ISSUE THREAD

**Issue by illwieckz** [DaemonEngine/Daemon#284]
As seen in #283, server list refreshing is done within its own loop. So either we manage to put that network thing in a separate thread, either we have to rewrite that thing in a way the main loop iterates the servers.

**Comment by illwieckz** [comment]
The `cl_pinglist` array is already a global variable, so we can keep the loop to just fill that array, and we may move this call:

```
Net::OutOfBandPrint(·netsrc_t::NS_CLIENT,·cl_pinglist[·j·].adr,·"getinfo·xxx"·);
```

Somewhere in the main loop to iterate it in the main loop.

We may then have to keep somewhere the time spent since the last query to avoid the issue described in #283.

**Comment by slipher** [comment]
The UI is not actually blocked(*)... it just chooses not to display any results until all pings have completed or timed out. This can be demonstrated by pressing Esc to close the list before the results show up. The cgame calls `trap_LAN_UpdateVisiblePings` each frame until it returns false, which indicates that pinging is finished.

I guess someone thought it would be annoying if the list moves around under your cursor or something. Maybe we don't have the ability to incrementally add items without destroying/recreating the existing items and messing up the selection state etc.

(*) Except by DNS queries. When the master server IPs are looked up for server listing purposes, the DNS queries block the engine's main thread. This part hasn't been ported to use the async DNS API.

**Comment by illwieckz** [comment]
Interesting? What would be good would be to not block the mouse pointer, even if the list is populated at once at the end of the fetch.

**Comment by slipher** [comment]
The mouse pointer is not blocked. But anyway we should populate it incrementally so that a slow pinging feature to fix #654 would not result in excessive waiting.
